src/chapters/objects-and-math/codingwithobjects.rst

```diff
@@ -0,0 +1,163 @@
+Coding With Objects
+===================
+
+Booleans and Objects
+--------------------
+
+Objects are not stored by their properties or by value, but by reference.
+Storing something by reference means that it is stored based on its location in memory.
+This can lead to some confusion when comparing objects.
+
+.. admonition:: Example
+
+   Let's see how this affects our zoo software! Surely, the zoo has more than one tortoise. The second tortoise is named Patricia!
+
+   .. sourcecode:: js
+      :linenos:
+
+      let tortoiseTwo = {
+          species: "Galapagos Tortoise",
+          name: "Patricia",
+          weight: 800,
+          age: 85,
+          diet: ["pumpkins", "lettuce", "cabbage"],
+          sign: function() {
+              return this.name + " is a " + this.species;
+          }
+       };
+
+   Because Pete and Patricia are members of the same species, are the same age, and have the same diet, you might notice that many of their properties are equal, but some are not.
+   Pete weighs more than Patricia and of course, they have different names!
+
+   For this example, we will only keep the ``species`` and ``diet`` properties.
+
+   .. sourcecode:: js
+      :linenos:
+
+      let tortoiseOne = {
+          species: "Galapagos Tortoise",
+          diet: ["pumpkins", "lettuce", "cabbage"]
+       };
+
+       let tortoiseTwo = {
+          species: "Galapagos Tortoise",
+          diet: ["pumpkins", "lettuce", "cabbage"],
+       };
+
+       console.log(tortoiseOne === tortoiseTwo);
+
+   **Console Output**
+
+   ::
+
+      False
+
+   The objects contain properties that have the same keys and equal values.
+   However, the output is ``False``.
+
+Even though ``tortoiseOne`` and ``tortoiseTwo`` have the same keys and values, they are stored in separate locations in memory.
+This means that even though you can compare the properties in different objects for equality, you cannot compare the objects themselves for equality.
+
+Iterating Through Objects
+-------------------------
+
+We can iterate through all of the values in an object, much like we would do with an array.
+We will use a ``for`` loop to do just that!
```

Comment:
To address your comment about too many "for" words, how about restructuring the three sentences like:
"We will use a ``for`` loop to do that, but with a slightly different structure.  ``for...in`` loops are specifically designed to loop through the properties of an object."

Reply:
Ok!